zipline/test_algorithms.py

```diff
@@ -368,10 +377,17 @@ def handle_data(self, data):
             assert self.portfolio.positions[0]['last_sale_price'] == \
                 data[0].price, "Orders not filled at current price."
 
-        self.order_percent(0, .001)
-        self.target_shares += np.floor((.001 *
-                                        self.portfolio.portfolio_value) /
-                                       data[0].price)
+        self.order_percent(self.sid(0), .001)
+
+        if isinstance(self.sid(0), Equity):
+            self.target_shares += np.floor(
+                (.001 * self.portfolio.portfolio_value) / data[0].price
+            )
+        if isinstance(self.sid(0), Future):
```

Comment:
I only did it as assets vs futures so that the same test algorithm could be used for multiple tests. A user would not have to make this kind of differentiation.